CMake 4.0 compatibility issue with deps/libimobiledevice/CMakeLists.txt

Describe the bug
On Arch Linux, which currently ships with CMake 4.0, building this project fails due to an outdated cmake_minimum_required() version in deps/libimobiledevice/CMakeLists.txt.

CMake 4.0 drops compatibility with very old versions (e.g. 2.8), and requires an updated minimum version to be declared explicitly.

To Reproduce
Steps to reproduce the behavior:

On Arch Linux (CMake 4.0), clone the repo.

Run:
"mkdir build && cd build
cmake .."

You get the following error:
"CMake Error at deps/libimobiledevice/CMakeLists.txt:2 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake."

Proposed fix
Update deps/libimobiledevice/CMakeLists.txt line 2 to:
"cmake_minimum_required(VERSION 3.5)"
or possibly higher (e.g. 4.0) if required by other parts of the codebase.

System info

OS: Arch Linux (rolling)

CMake version: 4.0.0

GCC: 15.1.1

Additional context
This issue may become more common as newer Linux distros adopt CMake ≥ 4.0.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -10,7 +10,7 @@ jobs:
   build-ubuntu:
     strategy:
       matrix:
-        os: [ubuntu-24.04, ubuntu-22.04, ubuntu-20.04]
+        os: [ubuntu-24.04, ubuntu-22.04]
     runs-on: ${{matrix.os}}
     steps:
     - name: Check Plugin sources
@@ -50,7 +50,8 @@ jobs:
         cp obs-ios-camera-source/data/locale/* package/.config/obs-studio/plugins/obs-ios-camera-source/data/locale
 
     - name: Upload artifacts
-      uses: actions/upload-artifact@v2
+      uses: actions/upload-artifact@v4
       with:
         name: plugin-${{matrix.os}}
         path: ${{runner.workspace}}/package
+        include-hidden-files: true

--- a/deps/libimobiledevice/CMakeLists.txt
+++ b/deps/libimobiledevice/CMakeLists.txt
@@ -1,5 +1,5 @@
 project(libimobiledevice)
-cmake_minimum_required(VERSION 3.2)
+cmake_minimum_required(VERSION 3.5)
 
 find_package(OpenSSL REQUIRED)
 

